### DISCUSSION TITLE
Let System Corruption Scan - in the first step - do only check - without fixing errors.

### DISCUSSION THREAD

**Discussion by ChrisNewbie** [ChrisTitusTech/winutil#2391] [Ideas]
about
functions/public/Invoke-WPFPanelDISM.ps1

I have a suggestion that the 'System Corruption Scan' option should have two stages: the first to check for errors and the second to repair them. And the repair should be done after user acceptance and as a result of an informed decision.

**Current options are:**
1. Chkdsk /Scan
2. SFC /ScanNow
3. DISM /Online /Cleanup-Image /Restorehealth
4. SFC /ScanNow

**My proposition:**
1. Chkdsk
2. SFC /VERIFYONLY
3. DISM /Online /Cleanup-Image /ScanHealth
_[and if user made decision:]_
4. Chkdsk /Scan
5. SFC /ScanNow
6. DISM /Online /Cleanup-Image /Restorehealth
7. SFC /ScanNow

**Comment by klei0742** [comment]
Also adding analzye component store, and if needed, clean store. otherwise you could just be worsening corruption by running these commands over time...

**Comment by richardgamarra** [comment]
Dism.exe /Online /Cleanup-Image /AnalyzeComponentStore
